src/DynamoCore/Library/XmlDocumentationExtensions.cs

```diff
@@ -199,6 +197,94 @@ private static string GetMemberElementName(FunctionDescriptor member)
             return String.Format("{0}:{1}", prefixCode, memberName);
         }
 
+        private enum XmlTagType
+        {
+            None,
+            Member,
+            Summary,
+            Parameter,
+            SearchTags
+        }
+
+        private enum DocumentElementType
+        {
+            Summary,
+            Description,
+            SearchTags
+        }
+
+        private static void LoadDataFromXml(XmlReader reader, string assemblyName)
+        {
+            if (reader == null)
+                return;
+
+            MemberDocumentNode currentDocNode = null;
+            XmlTagType currentTag = XmlTagType.None;
+            string currentParamName = String.Empty;
+
+            while (reader.Read())
+            {
+                switch (reader.NodeType)
+                {
+                    case XmlNodeType.Element:
+                        switch (reader.Name)
+                        {
+                            case "member":
+                                // Find attribute "name".
+                                if (reader.MoveToAttribute("name"))
+                                {
+                                    currentDocNode = new MemberDocumentNode(assemblyName, reader.Value);
+                                    documentNodes.Add(currentDocNode.FullyQualifiedName, currentDocNode);
+                                }
+                                currentTag = XmlTagType.Member;
+                                break;
+
+                            case "summary":
+                                currentTag = XmlTagType.Summary;
+                                break;
+
+                            case "param":
+                                if (reader.MoveToAttribute("name"))
+                                {
+                                    currentParamName = reader.Value;
+                                }
+                                currentTag = XmlTagType.Parameter;
+                                break;
+
+                            case "search":
+                                currentTag = XmlTagType.SearchTags;
+                                break;
+
+                            default:
+                                currentTag = XmlTagType.None;
+                                break;
+                        }
+                        break;
+                    case XmlNodeType.Text:
+                        if (currentDocNode == null)
+                            if (reader.Value == assemblyName)
+                                continue;
+                            else
+                                throw new InvalidDataException("Wrong XML documentation file.");
```

Comment:
On a second thought, I don't think we should throw an exception here, that's too harsh. Maybe a `break` or `continue` would do :)